src/index.ts

```diff
@@ -239,6 +272,16 @@ class Spanner extends GrpcService {
       },
       options || {}
     ) as {}) as SpannerOptions;
+    const emulatorHost = Spanner.getSpannerEmulatorHost();
+    if (
+      emulatorHost &&
+      emulatorHost.endpoint &&
+      emulatorHost.endpoint.length > 0
+    ) {
+      options.servicePath = emulatorHost.endpoint;
```

Comment:
What is the difference between `apiEndpoint` vs `servicePath ` in options? It seems that `apiEndpoint` has a higher priority and why don't we set `apiEndpoint` here instead of `servicePath`? 

https://github.com/googleapis/nodejs-spanner/blob/32fb6ca547b96fac1d742fc7580e78be069d2c02/src/index.ts#L286-L289

Reply:
To my understanding, the `apiEndpoint` option is intended as a manual override for the endpoint to use that can be specified by the user, and that should take precedence over any default settings. The `servicePath` is the option that can be used to override the default service endpoint of the client. By setting the value of `SPANNER_EMULATOR_HOST` on this option, we ensure that any user that has manually set a value for `apiEndpoint` will get the same behavior regardless of whether `SPANNER_EMULATOR_HOST` has been set.